File: solutions/php/difference-of-squares/1/DifferenceOfSquares.php
<?php

declare(strict_types=1);

/**
 * Square of the sum of the first $max natural numbers
 * Formula: (n * (n + 1) / 2)²
 */
function squareOfSum(int $max): int
{
    $sum = $max * ($max + 1) / 2;
    return (int) ($sum * $sum);
}

/**
 * Sum of squares of the first $max natural numbers
 * Formula: n * (n + 1) * (2n + 1) / 6
 */
function sumOfSquares(int $max): int
{
    return (int) ($max * ($max + 1) * (2 * $max + 1) / 6);
}

/**
 * Difference between square of sum and sum of squares
 */
function difference(int $max): int
{
    return squareOfSum($max) - sumOfSquares($max);
}

File: solutions/vbnet/difference-of-squares/1/DifferenceOfSquares.vb
Imports System.Linq

Public Module DifferenceOfSquares
    Public Function CalculateSquareOfSum(ByVal max As Integer) As Integer
        ' Sum of first n natural numbers: n * (n + 1) / 2
        Dim sum As Integer = max * (max + 1) / 2
        Return sum * sum
    End Function

    Public Function CalculateSumOfSquares(ByVal max As Integer) As Integer
        ' Sum of squares of first n natural numbers: n * (n + 1) * (2n + 1) / 6
        Return max * (max + 1) * (2 * max + 1) / 6
    End Function

    Public Function CalculateDifferenceOfSquares(ByVal max As Integer) As Integer
        Return CalculateSquareOfSum(max) - CalculateSumOfSquares(max)
    End Function
End Module